lib/mix/lib/mix/task.ex

```diff
@@ -215,6 +215,19 @@ defmodule Mix.Task do
     end
   end
 
+  @doc """
+  Gets requirements for the task.
+
+  Returns list with requirements (for example, `["compile"]`, or `["compile --warning-as-errors"]`), or `["compile", "app.start"]`.
+  """
+  @spec requirements(task_module) :: []
+  def requirements(module) when is_atom(module) do
+    case List.keyfind(module.__info__(:attributes), :requirements, 0) do
+      {:requirements, requirements} -> requirements
+      _ -> nil
+    end
```

Comment:
Since we're running through the requirements later, what if we go with defaulting to an empty list of requirements? I think that nicely signals that this task has no requirements.

```suggestion
    {:requirements, requirements} = List.keyfind(module.__info__(:attributes), :requirements, 0, {:requirements, []})
    requirements
```

Reply:
I like this idea!

Reply:
Great approach! :)